bindings/nodejs/lib/types/block/address.ts

```diff
@@ -118,12 +133,81 @@ class NftAddress extends Address {
     }
 }
 
+/**
+ * An implicit account creation address.
+ */
+class ImplicitAccountCreationAddress extends Address {
+    /**
+     * The Ed25519 address.
+     */
+    readonly address: Ed25519Address;
+    /**
+     * @param address An Ed25519 address.
+     */
+    constructor(address: Ed25519Address) {
+        super(AddressType.ImplicitAccountCreation);
+        this.address = address;
+    }
+
+    toString(): string {
+        return this.address.toString();
+    }
+}
+
+/**
+ * An address with restricted capabilities.
+ */
+class RestrictedAddress extends Address {
+    /**
+     * The inner address.
+     */
+    readonly address: Address;
+    /**
+     * The allowed capabilities bitflags.
+     */
+    allowed_capabilities: Uint8Array = new Uint8Array();
```

Comment:
I guess this will be a prefix hex string, if not there needs to be at least some conversion if we deserialize the type?

Reply:
According to Phillip, this will be an array of indexes when serialised

Reply:
Sorry he just corrected himself. It is indeed hexstring.
Here is an example `{"type":40,"address":{"type":0,"pubKeyHash":"0x4f172a5c322267154e25256a0c02702376107c3d593e7b67485b45706e4b752e"},"allowedCapabilities":"0x0203"}`